mcs/class/referencesource/mscorlib/system/threading/synchronizationcontext.cs

```diff
@@ -297,6 +297,11 @@ private static SynchronizationContext GetThreadLocalContext()
                 context = AndroidPlatform.GetDefaultSyncContext ();
 #endif
 
+#if UNITY_AOT
```

Comment:
Why only UNITY_AOT?

Reply:
I believe that the original change was only for IL2CPP. @TautvydasZilys do you recall?

Reply:
Indeed, this was specifically for Windows Runtime support on Xbox One and UWP, both of which are IL2CPP only.